bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetAutomationHandler.java

```diff
@@ -145,6 +149,30 @@ protected void requestChannelState(ChannelUID channel) {
             } catch (OWNException e) {
                 logger.debug("Exception while requesting channel {} state: {}", channel, e.getMessage(), e);
             }
+        } else {
+            logger.warn("Could not requestChannelState(): deviceWhere is null");
+        }
+    }
+
+    @Override
+    protected void refreshDevice(boolean refreshAll) {
+        OpenWebNetBridgeHandler brH = bridgeHandler;
+        if (brH != null) {
+            if (brH.isBusGateway() && refreshAll) {
+                long now = System.currentTimeMillis();
+                if (now - lastAllDevicesRefreshTS > ALL_DEVICES_REFRESH_INTERVAL_MSEC) {
+                    try {
```

Comment:
@fwolter I try to explain how the binding updates states. 
The binding is always connected to a BTicino gw using a Monitor TCP connection; when a light/shutter is physically operated the binding receives from the gw an event via Monitor TCP and it changes the channel via updateState. 
There is no polling and so no need of cached values. 

When the OH/binding restarts or connection to gw is disconnected->reconnected, we need to request all the lights/shutters states from gw.
In the case of BUS and lights/shutter this can be done via a single openwebnet "request all lights/shutters" command via TCP: `send(Automation.requestStatus(WhereLightAutom.GENERAL.value()));`

So what I implemented is: for every BUS light/shutter thing, request its state after reboot/reconnect; but if this "request all" command has been just sent by another Thing (checking the static `lastAllDevicesRefreshTS`), we skip and just wait for its answer from the Monitor connection. Basically the fist light/shutter sends the openwebnet "refresh all" command, and then all the others wait the new states to arrive from the gateway. See `OpenWebNetAutomationHandler.java::refreshDevice`.

So I do not understand how using the `ExpiringCache ` can improve this mechanism.

Reply:
This is a very common use case and that is exactly what `ExpiringCache` is designed for. As your responses are received asychronously, you would need to return null in the action lambda, but that wouldn't lead to the intended behavior of `ExpiringCache` or you need some quirks. So, you can leave it as is.

Reply:
ok, then I'll leave as it is now